src/modes/install.ts

```diff
@@ -39,7 +46,30 @@ export default async function(pkgs: PackageRequirement[]) {
         if (program.includes("{{")) continue
 
         const pkgstr = utils.pkg.str(pkg)
-        const exec = `exec pkgx +${pkgstr} -- ${program} "$@"`
+        if (UNSAFE) {
+          const parts = pkgstr.split("/")
+          parts.pop()
+          await Deno.mkdir(Path.home().join(`.cache/pkgx/envs/${parts.join("/")}`).toString(), {recursive: true})
+        }
+        //FIXME: doing `set -a` clears the args env
+        const exec = UNSAFE ? undent`
+            ARGS="$@"
+            pkgx_resolve() {
+              mkdir -p "$(dirname \\"$\{XDG_CACHE_DIR:-$HOME/.cache\}/pkgx/envs/${pkgstr}.env\\")"
+              pkgx +${pkgstr} 1>"$\{XDG_CACHE_DIR:-$HOME/.cache\}/pkgx/envs/${pkgstr}.env"
+              run
+            }
+            run() {
+              if [[ -e "$\{XDG_CACHE_DIR:-$HOME/.cache\}/pkgx/envs/${pkgstr}.env" && -e "$\{PKGX_HOME:-$HOME/.pkgx\}/${pkgstr}/v*/bin/${program}" ]]; then
+                set -a
+                source "$\{XDG_CACHE_DIR:-$HOME/.cache\}/pkgx/envs/${pkgstr}.env"
+                exec "$\{PKGX_HOME:-$HOME/.pkgx\}/${pkgstr}/v*/bin/${program}" "$ARGS"
```

Comment:
you should probably follow this with a `set +a` for proper termination.

Reply:
So, after sourcing, it should `set +a` (to preserve env)?

Reply:
i _believe_ that will return it to default behavior, yes.